crates/ruff_linter/src/rules/pyupgrade/rules/pep695/non_pep695_type_alias.rs

```diff
@@ -261,26 +261,28 @@ pub(crate) fn non_pep695_type_alias(checker: &Checker, stmt: &StmtAnnAssign) {
 
 /// Generate a [`Diagnostic`] for a non-PEP 695 type alias or type alias type.
 fn create_diagnostic(
-    source: &str,
-    stmt_range: TextRange,
+    checker: &Checker,
+    stmt: AnyNodeRef,
```

Comment:
I'd prefer using `&Stmt`a here if we always have a `stmt`. It encodes the constraint stronger (It otherwise leaves me wondering if `stmt` is guaranteed to be a `stmt` or if it is a left over from a refactor

Reply:
I switched to use `StmtRef`